extensions/git/src/git.ts

```diff
@@ -1287,6 +1288,10 @@ export class Repository {
 			args.push(treeish);
 		}
 
+		if (config.get('checkoutDetached')) {
+			args.push('--detach');
```

Comment:
Plz, correct me, but from what I see, you are using both `--track` and `--detach` options simultaneously, which is not allowed:

```
LANG=C git checkout -q --detach -t origin/master
fatal: '--detach' cannot be used with '-b/-B/--orphan'

git --version
git version 2.28.0
```

Am I missing something?

Reply:
Sorry, my mistake. Should be fixed with the last commit.